zkstack_cli/crates/zkstack/src/commands/autocomplete.rs

```diff
@@ -0,0 +1,48 @@
+use std::{
+    fs::File,
+    io::{BufWriter, Write},
+};
+
+use anyhow::Context;
+use clap::CommandFactory;
+use clap_complete::{generate, Generator};
+use common::logger;
+
+use super::args::AutocompleteArgs;
+use crate::{
+    messages::{msg_generate_autocomplete_file, MSG_OUTRO_AUTOCOMPLETE_GENERATION},
+    Inception,
+};
+
+pub fn run(args: AutocompleteArgs) -> anyhow::Result<()> {
+    let cmd = Inception::command();
+    let filename = format!(
+        "_{}_{}",
+        cmd.get_name(),
+        args.generator.to_string().to_ascii_lowercase()
+    );
+    let path = args.out.join(&filename);
+
+    logger::info(msg_generate_autocomplete_file(
+        path.to_str()
+            .context("the output file path is an invalid UTF8 string")?,
+    ));
+
+    let file = File::create(path).context("Failed to create file")?;
+    let mut writer = BufWriter::new(file);
+
+    generate_completions(args.generator, &mut writer)?;
+
+    logger::outro(MSG_OUTRO_AUTOCOMPLETE_GENERATION);
+
+    Ok(())
+}
+
+pub fn generate_completions<G: Generator>(gen: G, buf: &mut dyn Write) -> anyhow::Result<()> {
+    let mut cmd = Inception::command();
```

Comment:
nit: you get Inception twice.

Reply:
Addresse in https://github.com/matter-labs/zksync-era/pull/3097/commits/919db1f693d49d807595afd568cdbc10b7a76e31